### DISCUSSION TITLE
Is there a way to not autoconnect on refresh?

### DISCUSSION THREAD

**Discussion by madhawk1296** [rainbow-me/rainbowkit#894] [Q&A] [answered]
When I disconnect my wallet and refresh, it automatically reconnects. Is there anyway to stay disconnected even after refresh?

**Comment by aalmada** [comment]
Unfortunately that's by design: https://docs.metamask.io/guide/ethereum-provider.html#ethereum-isconnected

To not reconnect on refresh, the user has to go to the wallet and "disconnect".

The word "disconnect" is being used for two different meanings. When disconnecting on the website, it means to stop RPC communication with the wallet. When disconnecting on the wallet, it means to remove the website domain from the allow list. As long as the website domain is in the allow list, it will show as connected when it's refreshed.

Unfortunately, that's confusing for the user. I have a custom "connect" button that doesn't support "disconnect". It shows a tooltip when the user hovers that tells the user to go to the wallet if they really want to "disconnect" the website from the wallet.

**Reply by madhawk1296** [reply]
> Unfortunately that's by design: https://docs.metamask.io/guide/ethereum-provider.html#ethereum-isconnected
> 
> To not reconnect on refresh, the user has to go to the wallet and "disconnect".
> 
> The word "disconnect" is being used for two different meanings. When disconnecting on the website, it means to stop RPC communication with the wallet. When disconnecting on the wallet, it means to remove the website domain from the allow list. As long as the website domain is in the allow list, it will show as connected when it's refreshed.
> 
> Unfortunately, that's confusing for the user. I have a custom "connect" button that doesn't support "disconnect". It shows a tooltip when the user hovers that tells the user to go to the wallet if they really want to "disconnect" the website from the wallet.

When I use WAGMI directly instead of rainbowkit, it automatically saves when you disconnect or when you stay connected(even though metamask is always connected). This seems to be a bug on rainbowkit's side IMO.

https://wagmi.sh/react/client#autoconnect-optional

**Comment by ahsanfarooq3423** [comment]
Any update on this, please? Is there any way to disable auto-connect feature?

**Reply by lukaw3d** [reply, marked as answer]
You can set `<WagmiProvider reconnectOnMount={false} config={config}>`
instead of `<WagmiProvider config={config}>` from https://www.rainbowkit.com/docs/installation#wrap-providers
